vulnfeeds/cves/versions.go

```diff
@@ -613,10 +394,45 @@ func Commit(u string) (string, error) {
 
 	// TODO(apollock): add support for resolving a GitHub PR to a commit hash
 
+	// Support for resolving a Github tag to a commit hash
+	// example: https://github.com/redis/redis/releases/tag/6.2.17
+	if parsedURL.Host == "github.com" {
+		possibleCommitHash, err := resolveGitTag(parsedURL, u, gitSHA1Regex)
+		if possibleCommitHash != "" && err == nil {
+			return possibleCommitHash, nil
+		}
+	}
 	// If we get to here, we've encountered an unsupported URL.
 	return "", fmt.Errorf("Commit(): unsupported URL: %s", u)
 }
 
+func resolveGitTag(parsedURL *url.URL, u string, gitSHA1Regex *regexp.Regexp) (string, error) {
+	directory, tag := path.Split(parsedURL.Path)
+	normalizedTag, err := git.NormalizeVersion(tag)
+	if err == nil {
+		tag = normalizedTag
+	}
+
+	if strings.HasSuffix(directory, "tag/") {
+		maybeRepoURL, err := Repo(u)
+		if err != nil {
+			return "", err
+		}
+
+		normalizedTags, err := git.NormalizeRepoTags(maybeRepoURL, nil)
+		if err != nil {
+			return "", err
+		}
+		for t, nTag := range normalizedTags {
+			if tag == t && gitSHA1Regex.MatchString(nTag.Commit) {
+				return nTag.Commit, nil
+			}
+		}
+
+	}
+	return "", err
```

Comment:
If the URL is not a tag url, I think we should return an error rather than just empty string.